### PR TITLE
User favorites creation

### PR DESCRIPTION
**Pull request by allieIsOkay**
Adds route for creating favorites using an API key and a location.
Adds favorites and location models.

@TyMazey can you take a look at this one for me? I know I desperately need to get a lot of this logic out of the routes files, but i'd appreciate the feedback.

### CODE REVIEW

File: routes/api/v1/favorites.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  var locationName;
+  User.findOne({
+    where: {
+      api_key: req.body.api_key
+    },
+    include: 'locations'
+  })
+  .then(user => {
+    return new Promise((resolve, reject) => {
+      user ? resolve(user) : reject("Invalid API Key")
```
Comment (TyMazey) [inline routes/api/v1/favorites.js:20]: this syntax here for the promise looks really clean, i havnt seen that before but it looks smooth.

File: routes/api/v1/favorites.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      })
+
+      location ?
+      reject({error: req.body.location + " is already favorited"}) :
+      resolve(user)
+    })
+  })
+  .then(user => {
+    foundUser = user;
+    return findOrCreateCity(req.body.location);
```
Comment (TyMazey) [inline routes/api/v1/favorites.js:36]: like that you broke this fetch call function out

**Review by TyMazey — APPROVED** [review]
Overall looks really good, maybe a few spots you could break things into smaller functions, but as far as handling the request the flow is great

File: routes/api/v1/favorites.js (hunk 2 above)
Comment (allieIsOkay) [reply routes/api/v1/favorites.js:36]: Thanks, I'll try and move more stuff out.

**Pull request merged**